ssv/runner.go

```diff
@@ -234,9 +234,10 @@ func (b *BaseRunner) hasRunningDuty() bool {
 }
 
 func (b *BaseRunner) ShouldProcessDuty(duty *types.Duty) error {
-	if b.QBFTController.Height >= qbft.Height(duty.Slot) && b.QBFTController.Height != 0 {
+	// assume StartingDuty is not nil if state is not nil
+	if b.State != nil && b.State.StartingDuty.Slot >= duty.Slot {
```

Comment:
Why not just add b.QBFTController != nill check? 
Why change it to the above which changes the whole logic

Reply:
While fixing the issue I thought about making things better.

1. In general each validator duty runner can run only one duty at a time.
2. There is no real reason to make rule (1) dependent on QBFT controller. If we do separation of concerns and processes it is unclear why the runner has to even have access to the internal state of QBFT controller.
3. With my current change all duties (including the ones that aren't beacon duties) follow the exact same pattern. Because of the special exemptions we did to duties with no QBFT we ran into this issue.

Reply:
added a fallback that uses QBFTController just to get this approved

Reply:
From my perspective, indeed, the change makes module separation better and avoids the problem with the validator registration and exit duties